hack/e2e/run.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
```

Comment:
i want to avoid us maintaining 4 copies (ebs, fsx, efs, provider) of similar but not identical testing bash scripts. AT the moment this script just wraps ginkgo, do we need it?

Reply:
efs is a copy/paste of ebs. Are we willing to copy/paste it here too? https://github.com/kubernetes-sigs/aws-efs-csi-driver/tree/master/hack/e2e

Reply:
to avoid duplication of efforts wrt common testing stuff (create kops cluster, build image, test it, delete it) we can either:

1. Continue this copy/paste / git read-tree pattern, with ebs's hack/e2e tree as the canonical source.
2. Move ebs's hack/e2e tree into its own repo and use the kubernetes-csi/csi-release-tools git subtree pattern: https://github.com/kubernetes-csi/csi-release-tools
3. Design something else entirely from the ground up. (REALLY want to avoid this. As much as I dislike bash, it really is the best tool for the job. For example, if at some point we add kops to kubetest2 then it will be easy to shell out to kubetest2 instead of kops. csi-release-tools is mostly bash, gce pd's testing scripts are bash https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/tree/master/test, etc.)

Reply:
Sounds reasonable, but I like the CSI drivers depending on cloud provider rather than the other way around.  I can see that happening anyway in the future, and there's no reason for cloud provider to depend on any of the CSI repositories, so I'm fine with unifying as long as the source of truth is here, or a separate repo (but that's probably unnecessary).

Reply:
I take that back, we probably want a separate repo to avoid the dependencies having to import a bunch of go code along with the bash scripts that they want.

Reply:
Note that kOps already is using kubetest2. The runs against the EBS CSI driver can be seen here: https://testgrid.k8s.io/kops-misc#kops-scenario-aws-ebs-csi-driver

The script for launching cluster + running tests is here: https://github.com/kubernetes/kops/blob/master/tests/e2e/scenarios/aws-ebs-csi/run-test.sh

I imagine you could set up something more or less identical with minimal amount of boilerplate/duplication.

Reply:
yeah the scenarios script in kops repo don't look bad, I think the cloud provider repo should copy that approach @nckturner

for CSI drivers, I need specifically to test using eksctl, so unless I implement a kubetest2 wrapper around eksctl the CSI drivers are still going to use the hack/e2e scripts for foreseeable future.  https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/master/hack/e2e/eksctl.sh

for testing on EKS in general, https://github.com/aws/aws-k8s-tester implements kubetest2, but I haven't tried yet.

Anyway, I might be worrying too much about the potential for code duplication, I think basically we should converge on relying on kubetest2 as much as possible and any other boilerplate around can be easily scripted/minimized as you point out in those run-test.sh scripts